lib/wrappers/file-system.ts

```diff
@@ -1,16 +1,53 @@
 import * as fs from "fs";
-import { Extract } from "unzip";
+import * as path from "path";
+import * as yauzl from "yauzl";
+import * as util from "util";
+
+const access = util.promisify(fs.access);
+const mkdir = util.promisify(fs.mkdir);
 
 export class FileSystem {
 	public exists(path: string): boolean {
 		return fs.existsSync(path);
 	}
 
 	public extractZip(pathToZip: string, outputDir: string): Promise<void> {
-		return new Promise<void>((resolve, reject) => {
-			const stream = fs.createReadStream(pathToZip).pipe(Extract({ path: outputDir }));
-			stream.on("close", resolve);
-			stream.on("error", reject);
+		return new Promise((resolve, reject) => {
+			yauzl.open(pathToZip, { autoClose: true, lazyEntries: true }, (openError, zipFile) => {
+				if (openError) {
+					return reject(openError);
+				}
+
+				zipFile.on('entry', entry => {
+					const fn = <string>entry.fileName;
+					if (/\/$/.test(fn)) {
+						return zipFile.readEntry();
+					}
+
+					zipFile.openReadStream(entry, (openStreamError, stream) => {
+						if(openStreamError) {
+							return reject(openStreamError);
+						};
+
+						const filePath = `${outputDir}/${fn}`;
+
+						return createParentDirsIfNeeded(filePath)
+							.catch(createParentDirError => reject(createParentDirError))
+							.then(() => {
+								const outfile = createOutfile(filePath);
+								stream.once('end', () => {
```

Comment:
I personally prefer the following
```
const outfile = fs.createWriteStream(filePath)
```

`createOutfile` is used only here and has only one line code so it could be removed.

Reply:
Yeah, I agree, will fix.